subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/configurations/DefaultConfiguration.java

```diff
@@ -600,12 +600,8 @@ private ResolveState resolveToStateOrLater(final InternalState requestedState) {
                 // Error if we are executing in a user-managed thread.
                 throw new IllegalStateException("The configuration " + identityPath.toString() + " was resolved from a thread not managed by Gradle.");
             } else {
-                // We don't currently have mutable access to the project, so report a deprecation warning and then continue by attempting to acquire mutable access
-                DeprecationLogger.deprecateBehaviour("Resolution of the configuration " + identityPath.toString() + " was attempted from a context different than the project context. Have a look at the documentation to understand why this is a problem and how it can be resolved.")
-                    .willBeRemovedInGradle8()
-                    .withUserManual("viewing_debugging_dependencies", "sub:resolving-unsafe-configuration-resolution-errors")
-                    .nagUser();
-                return domainObjectContext.getModel().fromMutableState(p -> resolveExclusively(requestedState));
+                String docsLink = documentationRegistry.getDocumentationFor("viewing_debugging_dependencies", "sub:resolving-unsafe-configuration-resolution-errors");
+                throw new IllegalStateException(String.format("Resolution of the configuration " + identityPath.toString() + " was attempted from a context different than the project context. See: %s for more information.", docsLink));
```

Comment:
What do you think about this saying "current project context" instead of just "project context"?

Reply:
I think this is an improvement, but I'm not completely sure it's always meaningful to say current project, especially with the tooling API.  I'd say revisit the wording later and leave the message unchanged in this PR.